app/src/main/java/com/moviefiend/torian/moviefiend/MainActivity.java

```diff
@@ -1,13 +1,26 @@
 package com.moviefiend.torian.moviefiend;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
+import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
+
+public class MainActivity extends AppCompatActivity implements MovieListFragment.Listener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        MovieListFragment fragment = (MovieListFragment) getFragmentManager().findFragmentById(R.id.movie_list_fragment);
+        fragment.setListener(this);
```

Comment:
A fragment can have a destroyed activity if it's a retained fragment (and then `getActivity` will return null) but that's not the use case I'm referring to it here.

I'm referring to a use case where both the activity and fragment gets re-created (which will always happen unless the fragment is a retained fragment).
In this case depending on where you invoke `setListener` then the fragment may or may not have a listener when it gets re-created.

In your code there will never be an issue because you set the listener in the `Activity#onCreate` method. So if the activity and fragment gets re-created then `onCreate` method will be called and the fragment will be updated with the correct listener.

However, if for example, the fragment and listener are set due to a user event, like in the example I provided then that user-click callback will not be called again when the activity and fragment gets re-created resulting in a null listener in the fragment.
By having a `setListener` method as part of the API you allow that to happen.